File: algorithms/2974.minimum-number-game/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> numberGame(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());

        vector<int> arr;
        for (int i = 0; i < n; i += 2) {
            arr.emplace_back(nums[i + 1]);
            arr.emplace_back(nums[i]);
        }
        return arr;
    }
};

File: algorithms/2974.minimum-number-game/solution.rs
impl Solution {
    pub fn number_game(nums: Vec<i32>) -> Vec<i32> {
        let mut nums = nums;
        nums.sort_unstable();

        let mut arr: Vec<i32> = Vec::with_capacity(nums.len());
        for i in (0..nums.len()).step_by(2) {
            arr.push(nums[i + 1]);
            arr.push(nums[i]);
        }
        arr
    }
}
